### PR TITLE
Create figure showing the GRE images

### PR DESCRIPTION
**Pull request by dpapp86**
Added a GRE plotter. 

To avoid malarkey with the formatting, the code is repeated here for better use by more git-literate people:

%matplotlib inline
# Select subject to show
subject = 'sub-05'
os.chdir(os.path.join(path_data, subject, "anat"))

# Defining crop limits for resulting figure
ymin = 60
ymax = 340 

#Defining aspect ratio necessary for anatominally accurate figure pixelaspect=(ymax-ymin)/70
anatomyaspect=((ymax-ymin)*0.6)/(70*3)
aspectcorrection=pixelaspect/anatomyaspect

# Load CP for central slices
CP_GRE=nib.load(f"{subject}_acq-CP_T2starw.nii.gz")

# Defining mask based on the magnitude image intensity threshold cslice=CP_GRE.shape[0] // 2 +15 #shows the SC seg best

# Defining dynamic range
dynmin = 50 
dynmax = 400

# Create a figure with multiple subplots
fig, axes = plt.subplots(2, 4, figsize=(8, 5))
font_size = 14
axes=axes.flatten()

# Plot each GRE image, layout corresponding to the B1+ map figure for i,shim_mode in enumerate(shim_modes):
    # Load data
    GRE=nib.load(f"{subject}_acq-{shim_mode}_T2starw.nii.gz")
    GRE=GRE.get_fdata()[cslice, ymin:ymax,:]
    #Flipping right-left to have the same facig as on the B1+ maps
    GRE=np.flipud(GRE)
    # Plot
    splot=axes[i+1]
    im = splot.imshow((GRE.T), cmap='gray', origin='lower',vmin=dynmin,vmax=dynmax, aspect=aspectcorrection)#, interpolation='spline36')
    splot.set_title(shim_mode, size=font_size)
    splot.axis('off')
    
    # Delete unneeded first subplot
    axes[0].set_axis_off()
    

plt.tight_layout()
plt.subplots_adjust(wspace=0.1, hspace=0.05, right=0.9) plt.show()

### CODE REVIEW

File: content/index.ipynb [hunk 1]
```diff
@@ -1 +1,1683 @@
...
+{
```
Comment (jcohenadad) [inline content/index.ipynb:1]: the formatting drastically changes, making it hard to see the diff-- also, i am wondering if this is the latest file we should be modifying? with the neurolibre business happening in parallel i'm confused about the correct workflow-- @mathieuboudreau to the rescue 😅

**Comment by agahkarakuzu** [comment]
Usually saving notebooks with outputs end up creating these large diffs, esp. when the outputs differ. But overall comparing notebooks between commits is an issue, you can install https://github.com/marketplace/review-notebook-app to make diffs manageable. 

`index.ipynb` is indeed the one, I requested a preview build to see if changes were reflected.

**Comment by jcohenadad** [comment]
> Usually saving notebooks with outputs end up creating these large diffs, esp. when the outputs differ. 

indeed, but we always clean up outputs before commitings/pushing. In this case, it seems like the formatting of the notebook was altered for some reasons (maybe re-formatted by another software than that used to create it in the first place). In the past, we could easily spot difference without the need for a 3rd-party notebook diff checker (despite the annoying XML formatting), but now it is not possible.

**Comment by dpapp86** [comment]
Revisiting this:

The issue still remains. If I look at the relevant index.ipynb file, either online on the github interface, or logged in to github on my phone, it shows up as a single line, with 1 LOC. Meanwhile, downloading (and not even editing) the file renders it as human-readable, multi-line code, whether I open it in pycharm, sublime text, MSVS, or even vim. 
I am unsure what the proper way of addressing this, and adding my rather small changes, is.

**Comment by agahkarakuzu** [comment]
@dpapp86 now there are two branches here `-patch-1` and `-GRE_plot`, there's an Analysis.ipynb file, which is not included in the neurolibre table of contenst.

If you can provide the code snippet here to generate the plot and tell me where do you want it to appear, I am happy to do it in a clean branch from `main` and send a PR.

**Comment by jcohenadad** [comment]
thank you @agahkarakuzu -- let's make sure @mathieuboudreau sees this-- to avoid parallel work

**Comment by dpapp86** [comment]
Many thanks, @agahkarakuzu !

In **7     |     Paper figures**

After the cell titled _"Create figure of B1+ maps"_

**Add** a new cell with the title _"Create GRE signal intensity figure for one subject (sub-05)"_

The content of the cell is:

```
%matplotlib inline

#Select subject to show
subject = 'sub-05'
os.chdir(os.path.join(path_data, subject, "anat"))

#Defining crop limits for resulting figure
ymin = 60
ymax = 340

#Defining aspect ratio necessary for anatominally accurate figure 
pixelaspect=(ymax-ymin)/70 #70 slices
anatomyaspect=((ymax-ymin)*0.6)/(70*3) #in-plane resolution of 0.6mm, slice resolution of 3 mm
aspectcorrection=pixelaspect/anatomyaspect

#Load CP for central slices
CP_GRE=nib.load(f"{subject}_acq-CP_T2starw.nii.gz")

#Defining mask based on the magnitude image intensity threshold
cslice=CP_GRE.shape[0] // 2 +15 #shows the SC seg best

#Defining dynamic range
dynmin = 50
dynmax = 400

#Create a figure with multiple subplots
fig, axes = plt.subplots(2, 4, figsize=(8, 5))
font_size = 14
axes=axes.flatten()

#Plot each GRE image, layout corresponding to the B1+ map figure 
for i,shim_mode in enumerate(shim_modes):
# Load data
    GRE=nib.load(f"{subject}_acq-{shim_mode}_T2starw.nii.gz")
    GRE=GRE.get_fdata()[cslice, ymin:ymax,:]
    #Flipping right-left to have the same facig as on the B1+ maps
    GRE=np.flipud(GRE)
    # Plot
    splot=axes[i+1]
    im = splot.imshow((GRE.T), cmap='gray', origin='lower',vmin=dynmin,vmax=dynmax, aspect=aspectcorrection)#, interpolation='spline36')
    splot.set_title(shim_mode, size=font_size)
    splot.axis('off')

# Delete unneeded first subplot
axes[0].set_axis_off()
plt.tight_layout()
plt.subplots_adjust(wspace=0.1, hspace=0.05, right=0.9) 
plt.show()
```

**Sanity check**

The resulting figure should be this:
image

**Comment by agahkarakuzu** [comment]
@dpapp86 I assume Figure orders change in this case (3->4), also do you have the caption?

**Comment by dpapp86** [comment]
Good catch!

RF shimming figure (untocuhed, above the insert) is Figure 2

GRE Figure, the insert is Figure 3. Caption:
`Figure 3: GRE magnitude signal intensity for one participant (sub-05) across RF shimming conditions. The colomap adjustment is the same across all images. The differences in B1+ efficiency along the cord between the various RF shimming conditions results in signal intensity differences. Visual contrast between CSF and SC is affected, with ‘volume’ showing poor visual contrast compared to e.g. the ‘CP’ or ‘SAReff’ modes.`

Curves figures becomes Figure 4

**Comment by agahkarakuzu** [comment]
@dpapp86 I created the notebook and pushed it to a new branch, confirmed that it works on NeuroLibre. We can close this one and I'll open a new PR.

**Comment by dpapp86** [comment]
Hosannah! Thank you @agahkarakuzu !

**Comment by mathieuboudreau** [comment]
> thank you @agahkarakuzu -- let's make sure @mathieuboudreau sees this-- to avoid parallel work

✅ 🙂

**Pull request closed without merging**